node/src/accountant/dao_utils.rs

```diff
@@ -0,0 +1,479 @@
+        F2: Fn(&Row) -> rusqlite::Result<R>,
+        S: TryFrom<N>,
+        i128: From<N>,
+    {
+        let (finalized_stm, params): (String, RusqliteParamsWithOwnedToSql) = match self {
+            Self::TopRecords { count, ordered_by } => {
+                let (order_by_first_param, order_by_second_param) =
+                    Self::ordering(ordered_by, variant_top.age_param);
+                (
+                    stm_assembler(AssemblerFeeder {
+                        main_where_clause: variant_top.gwei_min_resolution_clause,
+                        where_clause_extension: "",
+                        order_by_first_param,
+                        order_by_second_param,
+                        limit_clause: variant_top.limit_clause,
+                    }),
+                    vec![(":limit_count", Box::new(count as i64))],
+                )
+            }
+            Self::RangeQuery {
+                min_age_s: min_age,
+                max_age_s: max_age,
+                min_amount_gwei: min_amount,
+                max_amount_gwei: max_amount,
+                timestamp,
+            } => (
+                stm_assembler(AssemblerFeeder {
+                    main_where_clause: variant_range.where_clause,
+                    where_clause_extension: variant_range.gwei_min_resolution_clause,
+                    order_by_first_param: "balance_high_b desc, balance_low_b desc",
+                    order_by_second_param: variant_range.secondary_order_param,
+                    limit_clause: "",
+                }),
+                Self::set_up_age_constrains(min_age, max_age, timestamp)
+                    .into_iter()
+                    .chain(Self::set_up_wei_constrains(vec![min_amount, max_amount]))
+                    .collect::<Vec<(&str, Box<dyn ToSql>)>>(),
+            ),
+        };
+        let accounts = Self::execute_query(conn, &finalized_stm, params, value_fetcher);
+        (!accounts.is_empty()).then_some(accounts)
+    }
+
+    fn execute_query<'a, R, F1>(
+        conn: &'a dyn ConnectionWrapper,
+        stm: &'a str,
+        params: RusqliteParamsWithOwnedToSql,
+        value_fetcher: F1,
+    ) -> Vec<R>
+    where
+        F1: Fn(&Row) -> rusqlite::Result<R>,
+    {
+        conn.prepare(stm)
+            .expect("select statement is wrong")
+            .query_map(
+                &*params
+                    .iter()
+                    .map(|(param_name, value)| (*param_name, value.as_ref()))
+                    .collect::<Vec<_>>(),
+                value_fetcher,
+            )
+            .unwrap_or_else(|e| panic!("database corrupt: {}", e))
+            .flatten()
+            .collect::<Vec<R>>()
+    }
+
+    fn set_up_age_constrains(
+        min_age: u64,
+        max_age: u64,
+        timestamp: SystemTime,
+    ) -> RusqliteParamsWithOwnedToSql {
+        let now = to_time_t(timestamp);
+        let to_time_t = |limit| now - checked_conversion::<u64, i64>(limit);
```

Comment:
Naming this closure `to_time_t` just below a call to the global `to_time_t` is confusing. Can you name it something that's more specific about exactly what it does? It looks like it converts a relative age to an absolute timestamp; maybe something like `age_to_timestamp` would be appropriate.